packages/integration-tests/src/relations/OneToManyCollection.test.ts

```diff
@@ -182,6 +182,21 @@ describe("OneToManyCollection", () => {
     expect(p1.authors.get.length).toEqual(0);
   });
 
+  it("removes deleted entities from other foreign key", async () => {
+    // Given an publisher with an author
+    const em = newEntityManager();
+    // And the publisher and authors are loaded
+    const p1 = newPublisher(em, { authors: [{}] });
```

Comment:
@stephenh - FYI: I couldn't figure out how to get `this.entity.__orm.data[this.fieldName]` to be the actual `Publisher` entity, and not `"p:1"` without using the `newPublisher` factory 🤔

If you have thoughts on how to repro this issue using `insertPublisher` and `em.load`, let me know and I can update.

Reply:
Yeah, I've pretty recently thought a similar "...wait, `data[fieldName]` is an entity" bugs I believe in either clone or changes.

I think the rub is that `data[fieldName]` is basically always an id/string _unless_ the `data[fieldName]` entity was brand new and didn't have an id, so "of course" needs to stay an object.

So, I think this is why these things "only show up when using factories" is that it's less the factory per se, and more about "oh this is a test that created the entities and, even if it's done an `em.flush`, is still using the same entities the test itself directly created".